ledger/sandbox/src/main/scala/com/digitalasset/platform/indexer/JdbcIndexer.scala

```diff
@@ -166,19 +159,18 @@ class JdbcIndexer private[indexer] (
     new SubscriptionResourceOwner(readService)
 
   private def handleStateUpdate(offset: Offset, update: Update): Future[Unit] = {
-    lastReceivedOffset = offset.toLedgerString
+    lastReceivedOffset = offset
     lastReceivedRecordTime = update.recordTime.toInstant
 
-    val externalOffset = Some(offset.toLedgerString)
+    val externalOffset = offset
     update match {
       case Heartbeat(recordTime) =>
         ledgerDao
           .storeLedgerEntry(
-            headRef,
-            headRef + 1,
             externalOffset,
-            PersistenceEntry.Checkpoint(LedgerEntry.Checkpoint(recordTime.toInstant)))
-          .map(_ => headRef = headRef + 1)(DEC)
+            PersistenceEntry.Checkpoint(LedgerEntry.Checkpoint(recordTime.toInstant))
+          )
+          .map(_ => ())(DEC)
```

Comment:
You can do this once, in the caller, rather than everywhere.

Reply:
With `in the caller` you mean change the return type of `LedgerDao#storeLedgerEntry` to `Future[Unit]`?
We indeed only use the result in tests to verify the behavior, but never in production code.

Reply:
I mean that this function can return `Future[PersistenceResponse]` and its caller, `SubscriptionResourceOwner#acquire`, can do the `.map(_ => ())` business.